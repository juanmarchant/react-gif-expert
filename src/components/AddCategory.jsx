import { useState } from "react"
import PropTypes from 'prop-types'
export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    //
    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();

        //Validamos para que no agreguen textos vacios
        if (inputValue.trim().length < 1) return;

        //Agregamos el valor con un callback para asi tener los valores anteriores y luego se ocupa un spread operator para agregarlos
        // setCategories(categories => [inputValue, ...categories]);
        
        onNewCategory(inputValue.trim());
        //Limpiamos
        setInputValue('');

    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}