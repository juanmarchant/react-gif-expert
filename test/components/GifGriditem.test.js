import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem"


describe('Haciendo Pruebas en GifGridItem', () => {

    const title = 'One Punch'
    const url = 'https://onepunch.com/onepunch.png'

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la img con el url y alt indicado', () => {
        render(<GifGridItem title={title} url={url} />);
        // screen.debug();
        
        // expect(screen.getByRole('img').src ).toBe(url)
        const {src, alt}=screen.getByRole('img');
        
        expect(src ).toBe(url)
        expect(alt ).toBe(title)
        

    });

    test('debe de mostrar el titulo en el componente', () => { 

        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy

     })

})