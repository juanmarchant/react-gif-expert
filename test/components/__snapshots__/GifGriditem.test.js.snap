// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo Pruebas en GifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://onepunch.com/onepunch.png"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
